testsuite/model/src/test/java/org/keycloak/testsuite/model/session/UserSessionProviderModelTest.java

```diff
@@ -296,20 +297,50 @@ public void testCreateUserSessionsParallel() throws InterruptedException {
                 (usMapStorageProvider == null || ConcurrentHashMapStorageProviderFactory.PROVIDER_ID.equals(usMapStorageProvider)));
 
         Set<String> userSessionIds = Collections.newSetFromMap(new ConcurrentHashMap<>());
+        CountDownLatch latch = new CountDownLatch(4);
+        CountDownLatch latchEnd = new CountDownLatch(4);
 
-        inIndependentFactories(4, 30, () -> withRealm(realmId, (session, realm) -> {
-            UserModel user = session.users().getUserByUsername(realm, "user1");
-            UserSessionModel userSession = session.sessions().createUserSession(realm, user, "user1", "", "", false, null, null);
-            userSessionIds.add(userSession.getId());
+        inIndependentFactories(4, 30, () -> {
+            withRealm(realmId, (session, realm) -> {
+                UserModel user = session.users().getUserByUsername(realm, "user1");
+                UserSessionModel userSession = session.sessions().createUserSession(realm, user, "user1", "", "", false, null, null);
+                userSessionIds.add(userSession.getId());
 
-            return null;
-        }));
+                latch.countDown();
+
+                return null;
+            });
+
+            // wait for other nodes to finish
+            try {
+                latch.await();
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+
+            assertThat(userSessionIds, Matchers.iterableWithSize(4));
 
-        assertThat(userSessionIds, Matchers.iterableWithSize(4));
+            // wait a bit to allow replication
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+
+            withRealm(realmId, (session, realm) -> {
+                userSessionIds.forEach(id -> Assert.assertNotNull(session.sessions().getUserSession(realm, id)));
+
+                latchEnd.countDown();
```

Comment:
Is this second latch necessary?

Reply:
My reasoning was to prevent deletion of sessions in the cluster when nodes are destroyed. Maybe I'm too careful here. I can try to remove it and let it test by GHA.

Reply:
https://github.com/martin-kanis/keycloak/actions/runs/2774186250 It looks also good without ending latch. I'll remove it.

Reply:
Sessions should not be removed when nodes are destroyed, if it is the case we have to fix it. I believe the sessions are removed only in the `cleanEnvironment` method. 

Btw. there seems to be some collision with Dominik's commit around the DB fix. Could you have a look at the commit message?

Reply:
Fixed